### ISSUE TITLE
How much RAM is needed to build an engine?

### ISSUE THREAD

**Issue by harryjulian** [NVIDIA/TensorRT-LLM#236]
I'm trying to build an engine for `falcon-7b-instruct` using the `nvcr.io/nvidia/tritonserver:23.10-trtllm-python-py3` container. I'm running on a g5.2xlarge with 32GB of RAM and an extra 16GB of swap. The Falcon `build.py` script keeps crashing my machine.

How much memory do I need in order to create an engine? Is there a heuristic relative to the size of the model?

**Comment by Luis-xu** [comment]
Hi, @harryjulian . I am using tensorrt_llm to deploy the llama-2-70B model. When I built the engine with FP32, it almost cost 1000GB of RAM. I modified the `weight.py` referring to [https://github.com/NVIDIA/TensorRT-LLM/issues/102#issuecomment-1780231895](https://github.com/NVIDIA/TensorRT-LLM/issues/102#issuecomment-1780231895),  and then when building the engine, the memory usage was greatly reduced, only about 600GB was needed.

I hope this reply can help you.
